src/actions/walletConnectActions.js

```diff
@@ -163,20 +161,6 @@ export const approveWalletConnectConnectorRequestAction = (peerId: string, chain
       return;
     }
 
-    const chain = chainFromChainId[chainId];
-    const requiresSmartWalletDeployment = !isDeployedOnChainSelector(getState())[chain];
```

Comment:
Should we keep some sort of check like this in the action as a defensive measure, wdyt?

Reply:
IMO at least in this case it's unnecessary since the button is disabled and writing it in action encourages others to do so in first place while we should show this in UI before user does any action and block him from going further. If in similar flows it reaches this kind of point – we should fix that user is blocked before he comes to action.